tests/js/parser-server-tests.js

```diff
@@ -0,0 +1,155 @@
+                    type: "test",
+                    sequence: [
+                        {
+                            funcName: "gpii.schema.parser.tests.server.testFieldLookup",
+                            args:     ["{testEnvironment}.gateKeeper.validator.parser", "escaped.json", ".['[x][x]'].description", "How do textual cross marks make you feel?"] // path, expected
+                        }
+                    ]
+                },
+                {
+                    name: "Testing deep lookup with dots and single quotes...",
+                    type: "test",
+                    sequence: [
+                        {
+                            funcName: "gpii.schema.parser.tests.server.testFieldLookup",
+                            args:     ["{testEnvironment}.gateKeeper.validator.parser", "escaped.json", ".['this.that']['th\'other'].description", "How do increasingly sloppy variable names make you feel?"] // path, expected
+                        }
+                    ]
+                },
+                {
+                    name: "Testing evolution and collapse of multiple failures within a single field...",
+                    type: "test",
+                    sequence: [
+                        {
+                            funcName: "gpii.schema.parser.tests.server.validateAndTest",
+                            args:     ["{testEnvironment}.gateKeeper.validator", "base.json", { required: true, password: "pass" }, { fieldErrors: { password: ["Password must be 8 or more characters, and have at least one uppercase letter, at least one lowercase letter, and at least one number or special character."]}}] // validator, schemaKey, content, expected
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+});
+
+fluid.defaults("gpii.schema.parser.tests.server.environment", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    events: {
+        constructEverything: null,
+        onSchemasUpdated:    null
+    },
+    listeners: {
+        onSchemasUpdated: {
+            funcName: "fluid.log",
+            args:     ["The test environment was indeed notified..."]
+        }
+    },
+    components: {
+        gateKeeper: {
+            type:         "fluid.component",
+            createOnEvent: "constructEverything",
+            options: {
+                components: {
+                    validator: {
+                        type: "gpii.schema.validator.server.hasParser",
+                        options: {
+                            schemaDir: schemaDir,
+                            components: {
+                                parser: {
+                                    options: {
+                                        listeners: {
+                                            "onSchemasUpdated.notifyEnvironment": {
+                                                func: "{testEnvironment}.events.onSchemasUpdated.fire"
```

Comment:
I collapsed the use of the gatekeeper by one level and used `distributeOptions` for the deep options setting.  I left the `schemaDir` option alone, as the syntax to use `distributeOptions` would have been longer in that case.